core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1855,7 +1855,11 @@ object SnappySession extends Logging {
         evaluatePlan(df, session, key.sqlText, key)
       }
     }
-    CacheBuilder.newBuilder().maximumSize(300).build(loader)
+    val cacheSize = Property.PlanCacheSize.getOption(SnappyContext.initConf) match {
```

Comment:
Can't you use conf from SnappyContext.globalSparkContext and get rid if this separate initConf? I think if you have planCache as lazy, then it should work fine. If you want to test this in unit test, then will probably need a separate unit test suite since this will remain once set even after stopping SparkContext. Or you can have planCache as volatile which is cleared on SnappyContext.stopSnappyContext and re-created on next access. Alternatively you can just do manual testing for this property.